lib/services/resource.go

```diff
@@ -111,6 +115,13 @@ func PreserveResourceID() MarshalOption {
 	}
 }
 
+func SkipCheckAndSetDefaults() MarshalOption {
```

Comment:
I would add a godoc here, and I would make sure to mention _why_ someone would want to skip this. It hasn't been something we've ever had to do before.